spec/services/participants/change_schedule/base_spec.rb

```diff
@@ -45,6 +45,58 @@ def matches_lead_provider?
         expect { subject.call }.to raise_error(ActionController::ParameterMissing)
       end
     end
+
+    context "when changing to schedule suitable for the course" do
+      let(:user) { profile.user }
+      let(:profile) { create(:mentor_participant_profile) }
+      let(:schedule) do
+        create(
+          :schedule,
+          type: "Finance::Schedule::ECFMentor",
```

Comment:
```
create(:ecf_mentor_schedule, #...)
```

Reply:
I've just pushed a commit to swap this out